benchmarks/src/main/scala/benchmark/MergeDataLoad.scala

```diff
@@ -0,0 +1,197 @@
+  }
+
+  def parse(args: Array[String]): Option[MergeDataLoadConf] = {
+    OParser.parse(argParser, args, MergeDataLoadConf())
+  }
+}
+
+class MergeDataLoad(conf: MergeDataLoadConf) extends Benchmark(conf) {
+
+  protected def targetTableFullName = s"`${conf.dbName}`.`target_${conf.tableName}`"
+
+  protected def dataSourceLocation: String = conf.sourcePath.getOrElse {
+    s"s3://devrel-delta-datasets/tpcds-2.13/tpcds_sf${conf.scaleInGB}_parquet/${conf.tableName}/"
+  }
+
+  /**
+   * Creates the target table and all source table configuration used in merge test cases.
+   */
+  def runInternal(): Unit = {
+    val dbName = conf.dbName
+    val dbLocation = conf.dbLocation(dbName, suffix = benchmarkId.replace("-", "_"))
+    val dbCatalog = "spark_catalog"
+
+    require(conf.scaleInGB > 0)
+    require(Seq(1, 3000).contains(conf.scaleInGB), "")
+
+    log(s"====== Creating database =======")
+    runQuery(s"DROP DATABASE IF EXISTS ${dbName} CASCADE", s"drop-database")
+    runQuery(s"CREATE DATABASE IF NOT EXISTS ${dbName}", s"create-database")
+
+    log(s"====== Creating target table =======")
+    loadTargetTable()
+    log(s"====== Creating source tables =======")
+    MergeTestCases.testCases.map(_.sourceTable).distinct.foreach(loadSourceTable)
+    log(s"====== Created all tables in database ${dbName} at '${dbLocation}' =======")
+
+    runQuery(s"USE $dbCatalog.$dbName;")
+    runQuery("SHOW TABLES", printRows = true)
+  }
+
+  /**
+   * Creates the target Delta table and performs sanity checks.
+   */
+  protected def loadTargetTable(): Unit = {
+    val dbLocation = conf.dbLocation(conf.dbName, suffix = benchmarkId.replace("-", "_"))
+    val location = s"${dbLocation}/${conf.tableName}/"
+    val sourceFormat = "parquet"
+
+    runQuery(s"DROP TABLE IF EXISTS $targetTableFullName", s"drop-table-$targetTableFullName")
+
+    runQuery(
+      s"""CREATE TABLE $targetTableFullName
+                 USING DELTA
+                 LOCATION '$location'
+                 SELECT * FROM `${sourceFormat}`.`$dataSourceLocation`
+              """, s"create-table-$targetTableFullName", ignoreError = true)
+
+    val sourceRowCount =
+      spark.sql(s"SELECT * FROM `${sourceFormat}`.`$dataSourceLocation`").count()
+    val targetRowCount = spark.table(targetTableFullName).count()
+    val targetFileCount =
+      spark.table(targetTableFullName).select(countDistinct("_metadata.file_path"))
+    log(s"Target file count: $targetFileCount")
+    log(s"Target row count: $targetRowCount")
+
+    assert(targetRowCount == sourceRowCount,
+      s"Row count mismatch: source table = $sourceRowCount, " +
+      s"target $targetTableFullName = $targetRowCount")
+  }
+
+  /**
+   * Creates a source table for the given source table configuration by sampling the target table.
```

Comment:
I find a bit confusing the source/target terminology. I guess it is because in some cases you refer to the source/target from a cloning perspective and others to source/target from a merge perspective? This needs some disambiguation. For example, here it does not make sense to sample the target table to create the source. It should be the other way around.